leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -187,24 +186,24 @@ private LeaderPingResult pingLeader(Optional<PaxosValue> maybeGreatestLearnedVal
     }
 
     private void proposeLeadershipAfter(Optional<PaxosValue> value) {
-        lock.lock();
-        try {
-            log.debug("Proposing leadership with value [{}]", SafeArg.of("paxosValue", value));
-            if (!isLatestRound(value)) {
-                // This means that new data has come in so we shouldn't propose leadership.
-                // We do this check in a lock to ensure concurrent callers to blockOnBecomingLeader behaves correctly.
-                return;
-            }
+        synchronized (lock) {
```

Comment:
why not synchronized on the method?

Reply:
External consumers can lock on the object, where nothing else can acquire this internal lock